### ISSUE TITLE
When generate chart with non-ascii char are broken.

### ISSUE THREAD

**Issue by tsuyoshicho** [GitbookIO/plugin-puml#1]
Thanks for useful plugin.

But I tried out with non-ascii generate chart. It is broken.

see below:
gitbook generate _broken_
![broken](http://www.plantuml.com/plantuml/svg/UDhZSlVLjGjmDbImKiX8pSd9rp-y-53ZeIumkHv8DXtgu080-RaGc000)

_manual_ generate via http://www.plantuml.com/plantuml/form
![manual](http://www.plantuml.com/plantuml/svg/S_VLjGjmDbImKiX8pSd9rt_SsFMuSVtZfWu0)

original

<pre>
{% plantuml %}
GM->PC1 : hello/おはよ
{% endplantuml %}
</pre>

---

Generation process code is

```
                var encoded = plantumlEncoder.encode(block.body);
                var href = 'http://www.plantuml.com/plantuml/' + format + '/' + encoded;
```

[plantuml-encoder](https://github.com/markushedvall/plantuml-encoder)' API is

```
module.exports.encode = function (text) {
  var utf8 = unescape(encodeURIComponent(text))
  var deflated = pakoDeflate.deflate(utf8, { level: 9, to: 'string' })
  return encode64.encode(deflated)
}
```

My gitbook is written as asciidoc at UTF-8.
block.body is not UTF-8?

Please fix this problem.

Regard.
CHO

**Comment by tsuyoshicho** [comment]
I tried debugging (on desk/source). then open  plantuml-encoder issue.

**Comment by MienDev** [comment]
+1,
 I run into the same trouble with that. 
(chinese character is not supported.)

**Comment by tsuyoshicho** [comment]
[plantuml-encoder#4](https://github.com/markushedvall/plantuml-encoder/issues/4#issuecomment-230163630) was closed.

This related issue too.

**Comment by tohosokawa** [comment]
I attempted to use this plugin with non-ascii letters today.

But they were broken.

I think that the cause would be "plantuml-encoder": "1.2.3" in package.json.

plantuml-encoder was fixed at 1.2.4.
